esdt-safe/src/lib.rs

```diff
@@ -335,15 +343,28 @@ pub trait EsdtSafe:
                 *total += &actual_bridged_amount;
             });
         }
-        self.create_transaction_event(
-            batch_id,
-            tx_nonce,
-            payment_token,
-            actual_bridged_amount,
-            required_fee,
-            user_addr.as_managed_buffer().clone(),
-            tx.to,
-        );
+        if !is_refund_tx {
+            self.create_transaction_event(
+                batch_id,
+                tx_nonce,
+                payment_token,
+                actual_bridged_amount,
+                required_fee,
+                refund_info.address.as_managed_buffer().clone(),
+                tx.to,
+            );
+        } else {
+            self.create_refund_transaction_event(
+                batch_id,
+                tx_nonce,
+                payment_token,
+                actual_bridged_amount,
+                required_fee,
+                refund_info.initial_batch_id,
+                refund_info.initial_nonce,
```

Comment:
I think we also need to include the refund address and tx.to in the create_refund_transaction_event because if we do not receive the create_transaction_event before, we don't have all the details to index the "refunded" swap in ES